resolwe/flow/views/mixins.py

```diff
@@ -102,7 +103,55 @@ def update(self, request, *args, **kwargs):
                         'Invalid descriptor_schema slug "{}" - object does not exist.'.format(ds_query)]},
                     status=status.HTTP_400_BAD_REQUEST)
 
-        return super(ResolweUpdateModelMixin, self).update(request, *args, **kwargs)
+        # return super(ResolweUpdateModelMixin, self).update(request, *args, **kwargs)
+        return self._update(request, *args, **kwargs)
+
+    # NOTE: This is a copy of rest_framework.mixins.UpdateModelMixin.update().
+    #       The self.get_object() was replaced with the
+    #       self.get_object_with_lock() to lock the object while updating.
+    #       Use the original method when suport for locking is added:
+    #       https://github.com/encode/django-rest-framework/issues/4675
+    def _update(self, request, *args, **kwargs):
+        """Update a resource."""
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object_with_lock()
```

Comment:
I think it would be nice to explicitly mark the changed line, even though this is mentioned in the comment above.